packages/@react-native-windows/telemetry/src/utils/errorUtils.ts

```diff
@@ -0,0 +1,157 @@
+
+  // autolink-windows
+  NoWindowsConfig: 3000,
+  IncompleteConfig: 3001,
+  InvalidConfig: 3002,
+  NeedAutolinking: 3003,
+  AddProjectToSolution: 3004,
+  Autolinking: 3005,
+};
+
+export type CodedErrorType = keyof typeof CodedErrors;
+
+/**
+ * Represents an error whose message might contain user-originating content,
+ * therefore when transmitting telemetry, only the type should be sent.
+ * @param type a stable ID identifying the type of error.
+ * @param message the error text. This should only be used for display to the user.
+ * @param data any additional metadata that is safe to collect for telemetry purposes.
+ */
+export class CodedError extends Error {
+  constructor(
+    public readonly type: CodedErrorType,
+    message: string,
+    public readonly data?: Record<string, any>,
+  ) {
+    super(message);
+    this.name = type;
+  }
+}
+
+/**
+ * Tries to parse an error code out of an error message.
+ * @param msg An error message to process.
+ * @returns The parsed error code.
+ */
+export function tryGetErrorCode(msg: string): string | undefined {
+  const errorRegEx = /error (\w+\d+):/gi;
+  const m = errorRegEx.exec(msg);
+  return m ? m[1] : undefined;
+}
+
+/**
+ * Sanitize an error message by anonymizing any paths that appear between quotes (''), brackets ([]), or double quotes ("").
+ * @param msg The error message to sanitize.
+ * @return The message with any paths anonymized.
+ */
+export function sanitizeErrorMessage(msg: string): string {
+  const cpuThreadId = /^\d+(:\d+)?>/g;
+  msg = msg.replace(cpuThreadId, '');
+  const parts = msg.split(/['[\]"]/g);
+  const clean = [];
+  const pathRegEx = /(['["]?)([A-Za-z]:|\\)[\\/]([^<>:;,?"*\t\r\n|/\\]+[\\/])+([^<>:;,?"*\t\r\n|]+\/?(['["]?))/gi;
+  for (const part of parts) {
+    if (pathRegEx.test(part)) {
+      pathRegEx.lastIndex = -1;
+      let matches: RegExpExecArray | null;
+      let noPath = '';
+      let last = 0;
+      while ((matches = pathRegEx.exec(part))) {
+        noPath +=
+          part.substr(last, matches!.index - last) +
+          sanitizeUtils.getAnonymizedPath(matches[0]);
+        last = matches!.index + matches![0].length;
+      }
+      if (noPath !== '') {
+        clean.push(noPath);
+      }
+    } else if (part !== '') {
+      clean.push(part);
+    }
+  }
+  return clean.join('').trim();
+}
+
+/**
+ * Sanitizes an error stack frame.
+ * @param frame
+ */
+export function sanitizeErrorStackFrame(
+  frame: appInsights.Contracts.StackFrame,
+): void {
+  const parens = frame.method.indexOf('(');
+  if (parens !== -1) {
+    // case 1: method === 'methodName (rootOfThePath'
+    frame.method = frame.method.substr(0, parens).trim();
+  } else {
+    // case 2: method === <no_method> or something without '(', fileName is full path
+  }
+  // preserve only the last_directory/filename
+  frame.fileName = path.join(
+    path.basename(path.dirname(frame.fileName)),
+    path.basename(frame.fileName),
+  );
+  frame.assembly = '';
+}
```

Comment:
I do not think it is safe to include filename, since folks can hook other code into the CLI.

Maybe we can whitelist files in our package, relative to node_modules?

Reply:
I will just change this to use our more robust logic around anonymized paths.